sherlock/internal/api/sherlock/service_alerts_procedures_v3_sync.go

```diff
@@ -0,0 +1,197 @@
+		json_bytes, err := createSvcAlertJsonData(ctx, alerts)
+		if err != nil {
+			errors.AbortRequest(ctx, fmt.Errorf("issue encountered creating json payload: %v", err))
+			return
+		}
+		// Upload file to bucket w/ latest info
+		if err = gcsClient.WriteBlob(ctx, *gcs_bucket, "alerts.json", json_bytes); err != nil {
+			errors.AbortRequest(ctx, fmt.Errorf("error writing updated alerts.json file: %v", err))
+			return
+		}
+	}
+}
+
+func getAlerts(ctx *gin.Context, request ServiceAlertV3SyncRequest, db *gorm.DB) ([]models.ServiceAlert, *string) {
+	var env_result models.Environment
+	if request.OnEnvironment != nil {
+		// match env so that we can get gcs bucket
+		environmentQuery, err := environmentModelFromSelector(*request.OnEnvironment)
+		if err != nil {
+			errors.AbortRequest(ctx, fmt.Errorf("error parsing environment selector '%s': %w", *request.OnEnvironment, err))
+			return nil, nil
+		}
+		if err = db.Where(&environmentQuery).Select("id").First(&env_result).Error; err != nil {
+			errors.AbortRequest(ctx, fmt.Errorf("error fetching environment '%v'", err))
+			return nil, nil
+		}
+	}
+	var activeAlerts []models.ServiceAlert
+	// Only return service alerts that haven't been deleted for this environment
+	if err := db.Model(&models.ServiceAlert{}).Where("DeletedAt = '' AND OnEnvironmentID = '%v'", env_result.ID).Find(&activeAlerts).Error; err != nil {
+		errors.AbortRequest(ctx, fmt.Errorf("(%s) error querying for Service Alerts: %w", errors.InternalServerError, err))
+		return nil, nil
+	}
+	return activeAlerts, env_result.ServiceBannerBucket
+
+}
+
+func getFileJson(ctx *gin.Context, gcs_client *google_bucket.GcsClientActual, blob *storage.ObjectAttrs) []ServiceAlertJsonData {
+	byte_data, read_err := gcs_client.ReadBlob(ctx, blob)
+	if read_err != nil {
+		// handle error
+		return nil
+	}
+	var json_data []ServiceAlertJsonData
+	err := json.Unmarshal(byte_data, &json_data)
+	if err != nil {
+		// handle error
+		return nil
+	}
+	return json_data
+}
+
+func compareAlerts(ctx *gin.Context, json_slice []ServiceAlertJsonData, db_alerts []models.ServiceAlert) ([]ServiceAlertJsonData, []models.ServiceAlert, []ServiceAlertJsonData) {
+	// helper func, returns true if there are any differences between alert json and info from DB
+	alertsNeedUpdate := func(jsonAlert ServiceAlertJsonData, dbAlert models.ServiceAlert) bool {
+		return jsonAlert.Title != *dbAlert.Title ||
+			jsonAlert.Message != *dbAlert.AlertMessage ||
+			jsonAlert.Link != *dbAlert.Link ||
+			jsonAlert.Severity != *dbAlert.Severity
+	}
+
+	var active_alerts int = len(json_slice)
+	var alerts_in_db int = len(db_alerts)
+
+	if active_alerts == 0 && alerts_in_db == 0 {
+		errors.AbortRequest(ctx, fmt.Errorf("(%s) Nothing to do, no active service alerts and nothing to modify", errors.NotFound))
+		return nil, nil, nil
+	}
+
+	jsonMap := make(map[string]ServiceAlertJsonData)
+	dbMap := make(map[string]models.ServiceAlert)
+
+	var toAdd []ServiceAlertJsonData
```

Comment:
at 40 lines, I'd start to think about breaking out parts of this method. i.e. I'd probably want to break out add/remove/update even if it's just so you can test each feature seperately in it's own method.